hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflow.java

```diff
@@ -29,6 +29,7 @@ public interface QueryWorkflow {
      *
      * @param requestBuffer The raw protobuf query bytes. Must be a {@link Query} object.
      * @param responseBuffer The raw protobuf response bytes.
+     * @param shouldCharge If the workflow should charge for handling the query.
      */
-    void handleQuery(@NonNull Bytes requestBuffer, @NonNull BufferedData responseBuffer);
+    void handleQuery(@NonNull Bytes requestBuffer, @NonNull BufferedData responseBuffer, boolean shouldCharge);
```

Comment:
yeah agree with Michael, having two workflows makes it clear.

Reply:
Will be addressed as mentioned [above](https://github.com/hashgraph/hedera-services/pull/15554#discussion_r1782821754).